src/EditorFeatures/Core.Wpf/Suggestions/FixAll/FixAllCodeFixGetFixesService.cs

```diff
@@ -0,0 +1,110 @@
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public FixAllCodeFixGetFixesService()
+        {
+        }
+
+        public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
+            => this;
+
+        public async Task<Solution> GetFixAllChangedSolutionAsync(FixAllContext fixAllContext)
+        {
+            var codeAction = await GetFixAllCodeActionAsync(fixAllContext).ConfigureAwait(false);
+            if (codeAction == null)
+            {
+                return fixAllContext.Solution;
+            }
+
+            fixAllContext.CancellationToken.ThrowIfCancellationRequested();
+            return await codeAction.GetChangedSolutionInternalAsync(cancellationToken: fixAllContext.CancellationToken).ConfigureAwait(false);
+        }
+
+        public async Task<ImmutableArray<CodeActionOperation>> GetFixAllOperationsAsync(
+            FixAllContext fixAllContext, bool showPreviewChangesDialog)
+        {
+            var codeAction = await GetFixAllCodeActionAsync(fixAllContext).ConfigureAwait(false);
+            if (codeAction == null)
+            {
+                return ImmutableArray<CodeActionOperation>.Empty;
+            }
+
+            return await FixAllGetFixesServiceHelper.GetFixAllOperationsAsync(
+                codeAction, fixAllContext.State.Project, fixAllContext.State.CorrelationId,
+                FunctionId.CodeFixes_FixAllOccurrencesPreviewChanges,
+                showPreviewChangesDialog, fixAllContext.CancellationToken).ConfigureAwait(false);
+        }
+
+        private static async Task<CodeAction> GetFixAllCodeActionAsync(FixAllContext fixAllContext)
+        {
+            using (Logger.LogBlock(
+                FunctionId.CodeFixes_FixAllOccurrencesComputation,
+                KeyValueLogMessage.Create(LogType.UserAction, m =>
+                {
+                    m[FixAllLogger.CorrelationId] = fixAllContext.State.CorrelationId;
+                    m[FixAllLogger.FixAllScope] = fixAllContext.State.Scope.ToString();
+                }),
+                fixAllContext.CancellationToken))
+            {
+                CodeAction action = null;
+                try
+                {
+                    action = await fixAllContext.FixAllProvider.GetFixAsync(fixAllContext).ConfigureAwait(false);
+                }
+                catch (OperationCanceledException)
+                {
+                    FixAllLogger.LogComputationResult(fixAllContext.State.CorrelationId, completed: false);
+                }
+                finally
+                {
+                    if (action != null)
+                    {
+                        FixAllLogger.LogComputationResult(fixAllContext.State.CorrelationId, completed: true);
+                    }
+                    else
+                    {
+                        FixAllLogger.LogComputationResult(fixAllContext.State.CorrelationId, completed: false, timedOut: true);
```

Comment:
timed out?

Reply:
do we mean 'canceled'?

Reply:
Again, all of this is existing code. Just renaming the type and splitting out some common code into a shared type led Github to show file as new code.